### PR TITLE
Update doc for decoupling of remote cluster state with remote backed data storage

### LINKED ISSUE #6729: [DOC] Update Remote Cluster State Documentation
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Remote cluster state can now be enabled for non-remote backed data storage cluster as well. 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
Issue: https://github.com/opensearch-project/OpenSearch/issues/9884
PR: https://github.com/opensearch-project/OpenSearch/pull/11858

### PR DESCRIPTION
**Pull request by soosinha**
### Description
Remote cluster state is now decoupled with remote backed data storage. Updating the doc to reflect the same

### Issues Resolved
Closes #6729

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by shwetathareja — APPROVED** [review]

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 1]
```diff
@@ -24,8 +24,10 @@ _Cluster state_ is an internal data structure that contains the metadata of the
...
 
 When the remote cluster state feature is enabled, the cluster metadata will be published to a remote repository configured in the cluster.
-Any time new cluster manager nodes are launched after disaster recovery, the nodes will automatically bootstrap using the latest metadata stored in the remote repository. 
-After the metadata is restored automatically from the latest metadata stored, and if the data nodes are unchanged in the index data, the metadata lost will be automatically recovered. However, if the data nodes have been replaced, then you can restore the index data by invoking the `_remotestore/_restore` API as described in the [remote store documentation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/index/).
+Any time new cluster manager nodes are launched after disaster recovery, the nodes will automatically bootstrap using the latest metadata stored in the remote repository. This way you get durability of metadata.
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:27]:
```suggestion
Any time new cluster manager nodes are launched after disaster recovery, the nodes will automatically bootstrap using the latest metadata stored in the remote repository. This provides metadata durability. 
```

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 2]
```diff
@@ -24,8 +24,10 @@ _Cluster state_ is an internal data structure that contains the metadata of the
...
 
 When the remote cluster state feature is enabled, the cluster metadata will be published to a remote repository configured in the cluster.
-Any time new cluster manager nodes are launched after disaster recovery, the nodes will automatically bootstrap using the latest metadata stored in the remote repository. 
-After the metadata is restored automatically from the latest metadata stored, and if the data nodes are unchanged in the index data, the metadata lost will be automatically recovered. However, if the data nodes have been replaced, then you can restore the index data by invoking the `_remotestore/_restore` API as described in the [remote store documentation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/index/).
+Any time new cluster manager nodes are launched after disaster recovery, the nodes will automatically bootstrap using the latest metadata stored in the remote repository. This way you get durability of metadata. 
+
+Note that remote cluster state can be enabled independent of remote backed data storage.
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:29]:
```suggestion
You can enable remote cluster state independently of remote-backed data storage.
{: .note}

```

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 3]
```diff
@@ -24,8 +24,10 @@ _Cluster state_ is an internal data structure that contains the metadata of the
...
+If durability of data is required then the remote backed data storage should also be enabled as specified in [remote store documentation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/index/).
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:30]:
```suggestion
If you require data durability, you must enable remote-backed data storage as described in the [Remote store documentation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/index/).
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @soosinha! A couple of minor tweaks.

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 4]
```diff
@@ -24,8 +24,12 @@ _Cluster state_ is an internal data structure that contains the metadata of the
...
 
 When the remote cluster state feature is enabled, the cluster metadata will be published to a remote repository configured in the cluster.
-Any time new cluster manager nodes are launched after disaster recovery, the nodes will automatically bootstrap using the latest metadata stored in the remote repository. 
-After the metadata is restored automatically from the latest metadata stored, and if the data nodes are unchanged in the index data, the metadata lost will be automatically recovered. However, if the data nodes have been replaced, then you can restore the index data by invoking the `_remotestore/_restore` API as described in the [remote store documentation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/index/).
+Any time new cluster manager nodes are launched after disaster recovery, the nodes will automatically bootstrap using the latest metadata stored in the remote repository. This provides metadata durability. 
+
+You can enable remote cluster state independently of remote-backed data storage.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:29]: Should "the" precede "remote"?

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 5]
```diff
@@ -24,8 +24,12 @@ _Cluster state_ is an internal data structure that contains the metadata of the
...
+{: .note}
+
+If you require data durability, you must enable remote-backed data storage as described in the [Remote store documentation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/index/).
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:32]:
```suggestion
If you require data durability, you must enable remote-backed data storage as described in the [remote store documentation]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/remote-store/index/).
```

**Review by natebower — COMMENTED** [review]
@soosinha @kolchfa-aws One comment and one change. Thanks!

**Pull request merged**